### PR TITLE
[RDG-41] create player mechanics

### PR DESCRIPTION
**Pull request by DaphneSmits**
## Describe your changes

I have created the `Agent` class and its subclass `Player`. The `Agent` class represents the common functionality of all possible agents inside the game (i.e. the `Player` and all `Enemy` instances). This way we can reuse functions that both will use, such as `move()`, `render()` and `reset()`. 

I have moved the main player functionality from `environment.py` to the new `Player` class. Functionally nothing changed, but this restructuring makes it easier to generalize and add new agents to the game.  

## List of tests added

Added the test file: `test_player.py`, containing the following tests:
- `test_initialization()`
- `test_reset()`
- `test_move()`
- `test_render()`

## Checklist before requesting a review
- [x] PR title contains ticket ID
- [x] If it is a core feature, I have added thorough tests.
- [x] I have included documentation

### CODE REVIEW

File: tests/test_player.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        self.assertEqual(self.player.color, (255, 255, 255))
+
+    def test_reset(self):
+        self.player.location = (200, 200)
+        self.player.reset()
+        self.assertEqual(self.player.location, (100, 100))
+
+    def test_move(self):
+        self.player.move(self.player.actions[3])  # Move right
+        self.assertEqual(self.player.location, (101, 100))
```
Comment (MatthijsLuijten) [inline tests/test_player.py]: @DaphneSmits Can you add some small docstring (can be a one-liner) what this function tests? Or do you think this is not needed and the function names say enough?

File: src/agent.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        pass
+
+    def render(self, screen: pygame.Surface) -> pygame.Rect:
+        """Render the agent as a circle on the screen.
+
+        Args:
+            screen (pygame.Surface): The surface to render the agent on.
+
+        Returns:
+            pygame.Rect: the created circle.
```
Comment (MatthijsLuijten) [inline src/agent.py:32]: `pygame.Rect` --> Should this be a circle?

File: src/agent.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        """
+        return pygame.draw.circle(screen, self.color, self.location, self.size)
+
+    def move(self, move: Tuple[int, int]) -> None:
+        """Move the agent by a specified amount.
```
Comment (MatthijsLuijten) [inline src/agent.py]: Maybe something like: `Move the agent given a direction.`

File: src/player.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        """Reset the player to its initial state."""
+        self.location = self.start_location
+
+    def render(self, screen: pygame.Surface) -> None:
+        """Render the player on the screen.
+
+        Args:
+            screen (pygame.Surface): The surface to render the player on.
+        """
+        return super().render(screen)
```
Comment (MatthijsLuijten) [inline src/player.py]: Do we have to specify this function here if it calls the `super()` function? I am not sure about this, but maybe we can just leave the whole `def render()` function out of the class? I think we only want to specify this function if we want a different behavior then specified in the parent class.

File: src/player.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    def move(self, move: Tuple[int, int]):
+        """Move the player based on the given move.
+
+        Args:
+            move (Tuple[int, int]): Move to perform.
+        """
+        super().move(move)
```
Comment (MatthijsLuijten) [inline src/player.py]: Same as above

File: src/environment.py [hunk 6]
```diff
@@ -3,26 +3,30 @@
 from pygame.surface import Surface
 from typing import Tuple
 
+from src.player import Player
+
 
 class Environment:
     def __init__(
-        self, screen: Surface, agent_size: int = 10, render_on: bool = False
+        self, screen: Surface, player_size: int = 10, render_on: bool = False
```
Comment (MatthijsLuijten) [inline src/environment.py:11]: In `main.py`, should line 12 (`env = Environment(screen=screen, render_on=True)`) be updated with the player size as well?

**Review by MatthijsLuijten — CHANGES_REQUESTED** [review]
Great work, overall looks really good. Some minor comments and suggestions <3.
I didn't have the opportunity to load the PR in vscode, just checked it in browser, so I will have to do that yet! But you can already start with this :)

File: src/agent.py (hunk 2 above)
Comment (DaphneSmits) [reply src/agent.py:32]: I looked it up and it actually returns the bounding rectangle: https://www.pygame.org/docs/ref/draw.html#pygame.draw.circle:~:text=the%20entire%20circle.-,Returns,will%20be%20truncated)%20and%20its%20width%20and%20height%20will%20be%200,-Return%20type

File: src/environment.py (hunk 6 above)
Comment (DaphneSmits) [reply src/environment.py:11]: I think that is only necessary if we do not want to use the default value. I would say for now that is fine? Wdyt?

File: src/player.py (hunk 4 above)
Comment (DaphneSmits) [reply src/player.py]: Good point! I have deleted them and that works just fine! :smile:

**Review by DaphneSmits — COMMENTED** [review]
Thanks for the review! I have addressed the comments and/or left you a comment about them

**Review by MatthijsLuijten — APPROVED** [review]
Nice work!

**Pull request merged**